### PR TITLE
R4 DocumentReference Patient Access documentation for read and search

### PR DESCRIPTION
**Pull request by Sanvenky**
Description
----
Updated fhir.cerner.com to add  Patient Access support for DocumentReference read and search capability..

PR Checklist
----
- [x] Screenshot(s) of changes attached before changes merged.
- [ ] Screenshot(s) of changes attached after changes merged and published.

### CODE REVIEW

**Comment by Sanvenky** [comment]
Before merge:

image
image
image
image
image
image
image

File: lib/resources/example_json/r4_examples_document_reference.rb [hunk 1]
```diff
@@ -4,168 +4,391 @@ module Cerner
...
+    }.freeze
+
+    R4_DOCUMENT_REFERENCE_PATIENT_ACCESS_BUNDLE ||= {
+      'resourceType': 'Bundle',
+      'id': '941ab54e9-4501-446a-9cee-6cb14c37dab3',
+      'type': 'searchset',
+      'link': [
+        {
+          'relation': 'self',
+          'url': 'https://fhir-open.cerner.com/r4/ec2458f2-1e24-41c8-b71b-0e701af7583d/DocumentReference?patient=12742400'
```
Comment (aki-mathur) [inline lib/resources/example_json/r4_examples_document_reference.rb:299]: Url should be of closed endpoint not open
https://fhir-ehr.cerner.com/r4/ec2458f2-1e24-41c8-b71b-0e701af7583d/DocumentReference?patient=12742400

File: lib/resources/example_json/r4_examples_document_reference.rb [hunk 2]
```diff
@@ -4,168 +4,391 @@ module Cerner
...
+        }
+      ],
+      'entry': [
+        {
+          'fullUrl': 'https://fhir-open.cerner.com/r4/ec2458f2-1e24-41c8-b71b-0e701af7583d/DocumentReference/197411797',
```
Comment (aki-mathur) [inline lib/resources/example_json/r4_examples_document_reference.rb:304]: Url should be of closed endpoint not open
https://fhir-ehr.cerner.com/r4/ec2458f2-1e24-41c8-b71b-0e701af7583d/DocumentReference?patient=12742400

File: lib/resources/example_json/r4_examples_document_reference.rb [hunk 3]
```diff
@@ -4,168 +4,391 @@ module Cerner
...
+          'resource': R4_DOCUMENT_REFERENCE_PATIENT_ACCESS
+        },
+        {
+          'fullUrl': 'https://fhir-open.cerner.com/r4/ec2458f2-1e24-41c8-b71b-0e701af7583d/DocumentReference/197411793',
```
Comment (aki-mathur) [inline lib/resources/example_json/r4_examples_document_reference.rb:308]: Url should be of closed endpoint not open
https://fhir-ehr.cerner.com/r4/ec2458f2-1e24-41c8-b71b-0e701af7583d/DocumentReference?patient=12742400

**Review by aki-mathur — CHANGES_REQUESTED** [review]

File: content/millennium/r4/documents/document-reference.md [hunk 4]
```diff
@@ -69,13 +69,22 @@ _Implementation Notes_
...
+    GET https://fhir-open.cerner.com/r4/ec2458f2-1e24-41c8-b71b-0e701af7583d/DocumentReference?patient=12742400
 
 #### Response
 
 <%= headers status: 200 %>
 <%= json(:R4_DOCUMENT_REFERENCE_BUNDLE) %>
 
+#### Patient Authorization Request
+
+    GET https://fhir-myrecord.cerner.com/r4/ec2458f2-1e24-41c8-b71b-0e701af7583d/DocumentReference?patient=12742400
```
Comment (aki-mathur) [inline content/millennium/r4/documents/document-reference.md:81]: This should be pointing to ehr
https://fhir-ehr.cerner.com/r4/ec2458f2-1e24-41c8-b71b-0e701af7583d/DocumentReference?patient=12742400

File: content/millennium/r4/documents/document-reference.md [hunk 5]
```diff
@@ -98,13 +107,22 @@ List an individual DocumentReference by its id:
...
+    GET https://fhir-open.cerner.com/r4/ec2458f2-1e24-41c8-b71b-0e701af7583d/DocumentReference/197411797
 
 #### Response
 
 <%= headers status: 200 %>
 <%= json(:R4_DOCUMENT_REFERENCE) %>
 
+#### Patient Authorization Request
+
+    GET https://fhir-myrecord.cerner.com/r4/ec2458f2-1e24-41c8-b71b-0e701af7583d/DocumentReference/197411797
```
Comment (aki-mathur) [inline content/millennium/r4/documents/document-reference.md:119]: This should be pointing to ehr
https://fhir-ehr.cerner.com/r4/ec2458f2-1e24-41c8-b71b-0e701af7583d/DocumentReference?patient=12742400

**Review by aki-mathur — CHANGES_REQUESTED** [review]

File: lib/resources/example_json/r4_examples_document_reference.rb (hunk 1 above)
Comment (Sanvenky) [reply lib/resources/example_json/r4_examples_document_reference.rb]: addressed

File: lib/resources/example_json/r4_examples_document_reference.rb (hunk 2 above)
Comment (Sanvenky) [reply lib/resources/example_json/r4_examples_document_reference.rb]: addressed

File: lib/resources/example_json/r4_examples_document_reference.rb (hunk 3 above)
Comment (Sanvenky) [reply lib/resources/example_json/r4_examples_document_reference.rb]: addressed

File: content/millennium/r4/documents/document-reference.md (hunk 4 above)
Comment (Sanvenky) [reply content/millennium/r4/documents/document-reference.md]: addressed

File: content/millennium/r4/documents/document-reference.md (hunk 5 above)
Comment (Sanvenky) [reply content/millennium/r4/documents/document-reference.md]: addressed

File: lib/resources/example_json/r4_examples_document_reference.rb [hunk 6]
```diff
@@ -4,168 +4,391 @@ module Cerner
...
+        'display': 'Portal, Portal'
       },
       'content': [
         {
           'attachment': {
             'contentType': 'application/pdf',
-            'url': 'https://fhir-open.cerner.com/r4/ec2458f2-1e24-41c8-b71b-0e701af7583d/Binary/XR-197292845',
-            'title': 'Height Weight Allergy Rule',
-            'creation': '2020-07-06T21:02:47.000Z'
+            'url': 'https://fhir-open.cerner.com/r4/ec2458f2-1e24-41c8-b71b-0e701af7583d/Binary/XR-197411797',
```
Comment (aki-mathur) [inline lib/resources/example_json/r4_examples_document_reference.rb:163]: This should also point to ehr

File: lib/resources/example_json/r4_examples_document_reference.rb [hunk 7]
```diff
@@ -4,168 +4,391 @@ module Cerner
...
+              'display': 'Portal, Portal'
             },
             'content': [
               {
                 'attachment': {
                   'contentType': 'application/pdf',
-                  'url': 'https://fhir-open.cerner.com/r4/ec2458f2-1e24-41c8-b71b-0e701af7583d/Binary/XR-197292833',
-                  'title': 'Height Weight Allergy Rule',
-                  'creation': '2020-07-06T20:37:35.000Z'
+                  'url': 'https://fhir-open.cerner.com/r4/ec2458f2-1e24-41c8-b71b-0e701af7583d/Binary/XR-197411793',
```
Comment (aki-mathur) [inline lib/resources/example_json/r4_examples_document_reference.rb:378]: This should also point to ehr

**Review by aki-mathur — CHANGES_REQUESTED** [review]

File: lib/resources/example_json/r4_examples_document_reference.rb (hunk 6 above)
Comment (Sanvenky) [reply lib/resources/example_json/r4_examples_document_reference.rb]: addressed

File: lib/resources/example_json/r4_examples_document_reference.rb (hunk 7 above)
Comment (Sanvenky) [reply lib/resources/example_json/r4_examples_document_reference.rb]: addressed

**Review by aki-mathur — APPROVED** [review]

**Review by vinuthnag — APPROVED** [review]

**Comment by VelocityKnight** [comment]
post merge screenshots:
Screen Shot 2020-12-17 at 4 30 40 PM
Screen Shot 2020-12-17 at 4 31 08 PM
Screen Shot 2020-12-17 at 4 31 24 PM
Screen Shot 2020-12-17 at 4 31 57 PM

**Pull request merged**